src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/CodeActionEndpointTest.cs

```diff
@@ -751,5 +757,82 @@ public override Task<IReadOnlyList<RazorCodeAction>> ProvideAsync(RazorCodeActio
                 return Task.FromResult<IReadOnlyList<RazorCodeAction>>(null);
             }
         }
+
+        private class MockLanguageServer : ClientNotifierServiceBase
```

Comment:
Up for debate, but I don't think this is a `Mock`.. I would call it a `TestLanguageServer`, and I would love it if we moved away from mocks in general to this sort of thing, because it at least has the compiler to help us when implementations change.

Reply:
You are your naming requirements. Is this the kind of world that razor lives in? So strict